ASTInfoGenerator/DeclParser/DeclParser.py

```diff
@@ -0,0 +1,382 @@
+from ASTInfoGenerator.DeclParser.DeclTypes.FieldDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.StructVarDecl import *
+from ASTInfoGenerator.DeclParser.DeclTypes.AnnotateAttr import *
+
+
+class DeclParser:
+    def __init__(self):
+        self.__param_index = None
+        self.__field_index = None
+
+    def _parse_function_decl(self, function_decl: str) -> Specifier:
+        # Assumption: functions that are extern are not relevant to our analysis
+        if function_decl.endswith("extern"):
+            return None
+
+        logout(f"function decl equals {function_decl}")
+
+        quote_index = function_decl.find("'")
+        # Assumption: FunctionDecls wrap in quotes ('...') the return type of the function & parameters, and
+        # right before the return type is the name of the function.
+        # we also assume the function name and return type are separated by a space.
+        # e.g., |-FunctionDecl 0x24161a8 <../test/test23/test23.h:27:1, line:28:57> col:5 does_something 'struct my_struct (struct my_struct)'
+        #                                                                                 ^func name     ^return type      ^parameters
+
+        start_index = quote_index - 2
+        # quote_index - 2 since we will be traversing FunctionDecl
+        # backward, starting at the last character of the name of the function,
+        # in order to get the full function name
+
+        cur_char = function_decl[start_index]
+        function_name = ""
+
+        while (cur_char != " "):
+            function_name = cur_char + function_name
+
+            start_index -= 1
+            cur_char = function_decl[start_index]
+
+        start_index = quote_index + 1
+        # quote_index + 1 since we assume this is where
+        # the return type is specified
+
+        cur_char = function_decl[start_index]
+        return_type = ""
+
+        while (cur_char != "("):
+            # we assume that parameter types are specified within
+            # the open-close parentheses
+            # e.g., 'int (int)' is a function that returns int with singular parameter int
+            # e.g., 'int ()' is a function that returns int with no parameters
+
+            # e.g., 'void (int, int, char *)' is a function that returns void with
+            # 3 parameters (in the order as it appears in C code), int, int, and char *
+
+            return_type += cur_char
+
+            start_index += 1
+            cur_char = function_decl[start_index]
+
+        logout(
+            f"function name = '{function_name}', return type = '{return_type}'")
+        return FunctionDecl(function_name, return_type)
+
+    def _parse_record_decl(self, record_decl: str):
+        record_decl_chunks = record_decl.split(" ")
+        struct_name = record_decl_chunks[len(record_decl_chunks) - 2]
+
+        return RecordDecl(struct_name)
+
+    def _parse_parm_var_decl(self, parm_var_decl: str, param_index: int):
```

Comment:
Nit: the inconsistency between uses of `parm` and `param` here bothers me a bit. I would find it easier to read this code if we use `param` everywhere.

Reply:
I will increase the usage of `param` but will keep `parm` in some places since the AST defines parameter declarations as `ParmVarDecl`s.